public/app/index.ts

```diff
@@ -19,16 +19,8 @@ if (window.nonce) {
 window.__grafana_app_bundle_loaded = true;
 
 import app from './app';
+import { notifyIfMockApiEnabled, potentiallySetupMockApi } from './mock-api-utils';
 
-const prepareInit = async () => {
-  if (process.env.frontend_dev_mock_api) {
-    return import('test/mock-api/worker').then((workerModule) => {
-      workerModule.default.start({ onUnhandledRequest: 'bypass' });
-    });
-  }
-  return Promise.resolve();
-};
-
-prepareInit().then(() => {
-  app.init();
-});
+potentiallySetupMockApi()
+  .then(() => app.init())
+  .then(() => notifyIfMockApiEnabled());
```

Comment:
rather than coupling this very tightly to the mock api, what do you think about making this more lifecycle-ey? feels similar to the discussion we were having with @samsch in daily a couple of days ago.

this could be something like:
```tsx
preInitTasks()
  .then(() => app.init())
  .then(() => postInitTasks())
```

and it would just so happen that right now:
```tsx
async function preInitTasks() {
  await potentiallySetupMockApi();
}

async function postInitTasks() {
  notifyIfMockApiEnabled();
}
```